### PR TITLE
Simplify some examples.

### PR DESCRIPTION
**Pull request by anntzer**
`vline_hline_demo`: Simpler data generation; use `subplots`; no need to
explicitly construct a blended transform.

`equal_aspect_ratio`: Removed; forcing equal aspect ratio when the x and
y axes have different units is pretty silly anyways, plus it does not
belong in `subplots_axes_and_figures`.  Instead, tweaked the last
example of `axis_equal_demo` to demonstrate some more related features.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/whats_new.rst if major new feature
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/pylab_examples/axis_equal_demo.py [hunk 1]
```diff
@@ -32,10 +30,8 @@
 
 plt.subplot(224)
 plt.plot(3*np.cos(an), 3*np.sin(an))
-plt.axis('equal')
-plt.axis([-3, 3, -3, 3])
-plt.plot([0, 4], [0, 4])
-plt.title('still equal after adding line', fontsize=10)
+plt.gca().set_aspect('equal', 'box')
```
Comment (dstansby) [inline examples/pylab_examples/axis_equal_demo.py]: Could also do with a `plt.subplots()` and removing all the `plt.subplot` calls?

Comment (anntzer) [inline examples/pylab_examples/axis_equal_demo.py]: done

File: examples/pylab_examples/vline_hline_demo.py [hunk 2]
```diff
@@ -7,26 +7,18 @@
...
 
-fig = plt.figure(figsize=(12, 6))
-vax = fig.add_subplot(121)
-hax = fig.add_subplot(122)
+fig, (vax, hax) = plt.subplots(1, 2, figsize=(12, 6))
 
 vax.plot(t, s + nse, '^')
 vax.vlines(t, [0], s)
-vax.vlines([1, 2], 0, 1, transform=btf(vax.transData, vax.transAxes), colors='r')
+vax.vlines([1, 2], 0, 1, transform=vax.get_xaxis_transform(), colors='r')
```
Comment (story645) [inline examples/pylab_examples/vline_hline_demo.py]: I feel like `vax.get_xaxis_transform()` would benefit from an explanatory comment of why it's in this code.

File: examples/pylab_examples/axis_equal_demo.py [hunk 3]
```diff
@@ -3,40 +3,32 @@
...
 
-This example is only interesting when run in interactive mode.
-
 """
 
-
 import matplotlib.pyplot as plt
 import numpy as np
 
 # Plot circle or radius 3
```
Comment (story645) [inline examples/pylab_examples/axis_equal_demo.py]: type: circle of radius 3

File: examples/pylab_examples/vline_hline_demo.py (hunk 2 above)
Comment (anntzer) [inline examples/pylab_examples/vline_hline_demo.py]: you get to write it then :)

File: examples/pylab_examples/axis_equal_demo.py (hunk 3 above)
Comment (anntzer) [inline examples/pylab_examples/axis_equal_demo.py]: fixed

File: examples/pylab_examples/vline_hline_demo.py (hunk 2 above)
Comment (story645) [inline examples/pylab_examples/vline_hline_demo.py]: I have no idea why it's there, hence why I'd like to see that comment.

Comment (anntzer) [inline examples/pylab_examples/vline_hline_demo.py]: done

**Pull request merged**